### ISSUE TITLE
PROCESS no longer runs successfully after using write_new_in_dat script

### ISSUE THREAD

**Issue by jonmaddock** [ukaea/PROCESS#709]
In GitLab by @kellis on May 17, 2018, 16:11

NB. I am running PROCESS without PLASMOD, but it is the PLASMOD branch. Commit 6b313b5d.

Working from the 'baseline 2018' file given to me by James, I made some changes with (sensible) feasible results. I then made the following changes:
``
iradloss = 0 (previously default value of 1)
coreradius = 1.0
coreradiationfraction = 1.0
``
Following this successful run, I used script 'write_new_in_dat.py'. I ran the resulting file, but with no successful result - PROCESS failed with iflag 6 after 2 VMCON iterations. 

Script write_new_in_dat appears to be doing what I expect, so the question is, why is PROCESS having difficulty with a solution which it has already found? 

Files attached:
[iradloss0.zip](https://git.ccfe.ac.uk/process/process/uploads/60fbdab29eb7fa118f1f4034e36350d0/iradloss0.zip)

(filename 'iradloss' is original; new_IN is the one written by write_new_in_dat)

@mkovari @jmorris-uk

**Comment by jonmaddock** [comment]
This problem (Process not being able to solve when the initial optimisation parameter vector is the previous solution) has been resolved by #3186, which ensures that model evaluations are stable before returning objective function, constraints and gradients back to the solver, and all values in the MFILE are stable before writing a final solution file. I believe this problem was caused by a single model evaluation at the solution vector at the end of an optimisation run (i.e. with a history close to the solution) and the same point evaluated for the first time (the start of the subsequent optimisation run) providing different values to the solver, hence different convergence results. As the models are now ensured to be idempotent before returning values, this can no longer happen.

I have tested this with the large tokamak regression test solution, which, when converted into a new input file, converges instantly at the same point, as expected.
